core/arch/arch_exports.h

```diff
@@ -1051,18 +1069,19 @@ fill_with_nops(dr_isa_mode_t isa_mode, byte *addr, size_t size);
 #    define PC_LOAD_ADDR_ALIGN 4
 
 #elif defined(RISCV64)
-/* FIXME i#3544: This can be 2B in C. */
+/* Instructions from C extensions are 2 bytes in size and we will never use this macro on
+ * these instructions. */
 #    define RISCV64_INSTR_SIZE 4
-/* FIXME i#3544: Not implemented */
-#    define FRAGMENT_BASE_PREFIX_SIZE(flags) RISCV64_INSTR_SIZE
-/* FIXME i#3544: Not implemented */
+#    define RISCV64_INSTR_COMPRESSED_SIZE 2
+#    define FRAGMENT_BASE_PREFIX_SIZE(flags) RISCV64_INSTR_SIZE * 2
```

Comment:
> is this meant to go with a different PR?

Yes, I'm trying to keep the PR size small, is this proper?

Reply:
I'm guessing you had a branch with multiple pieces and later split it up and it was easiest to split by file and just leave these couple of defines in this PR instead of pulling the changes out of this already-modified file.  Since the RISC-V code is all still a work in progress and we're filling in many things this seems ok to not have a perfect separation so let's leave it.